### ISSUE TITLE
[CoE Starter Kit - BUG] CustomControlDefaultConfig error - can't import Core Components solution

### ISSUE THREAD

**Issue by zellos13** [microsoft/coe-starter-kit#8913]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

Hi,

I am struggling with the import. I am having a problem with some CustomControlDefaultConfig component. I can't locate it and I don't know how to find it.

Could you please provide me with the information how to overcome this problem?

The problem:

_The uninstall operation will delete the base layer for the component 'CustomControlDefaultConfig' with id 'b3a6b6fb-09c9-4b89-a71f-b8a623be4188'. The operation cannot continue because there are other managed layers over the base layer. You can use the solution layers to find out which other solutions are blocking the operation._

I saw this post: https://github.com/microsoft/coe-starter-kit/issues/8245 but it's a different type of component.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.31

### What app or flow are you having the issue with?

CustomControlDefaultConfig

### What method are you using to get inventory and telemetry?

None

**Comment by Jenefer-Monroe** [comment]
I don't know what that is, but it sounds like its related to this product regression that is being resolved:

## Upgrade issues around table ids
 
Over the last two weeks we have started seeing issues that state : Cannot delete attribute: admin_extableid from Entity: admin_ExTable since the attribute is not a custom field
We see this for tables that are still active and who’s id is still there and expected to not be custom.
 
Here are some examples:
•	AI Builder Models: https://github.com/microsoft/coe-starter-kit/issues/8831
•	Audit Logs: https://github.com/microsoft/coe-starter-kit/issues/8870
•	Customized Email: https://github.com/microsoft/coe-starter-kit/issues/8848
•	App Catalog: https://github.com/microsoft/coe-starter-kit/issues/8862


## Update from product team in those:
![image](https://github.com/user-attachments/assets/f0917805-acd5-43be-b729-677f8f82a921)

**Comment by zellos13** [comment]
In which version this fix is included?

**Comment by Jenefer-Monroe** [comment]
I'm not sure. I've reached out to the product team but have not heard back. 
Assigning to Grant to press with them.

**Comment by shmcarth** [comment]
Please retest after this last weekend.  Which region is your environment in? One thing to confirm if you experience the error again is to import the solution using the 2 stage holding pattern for upgrade, then complete the upgrade.

**Comment by zellos13** [comment]
Hello,

Are you guys sure this is a problem with the CoE package? I didn't find this error before on the CoE Starter Kit GitHub page. 

Which region is your environment in? EUROPE
Please retest after this last weekend. When exactly should I retest, please tell me which version to use so this workaround is included.
After I am confident about which version I should use, I will import the solution using the 2-stage holding pattern for an upgrade.

We are discussing possible other solutions with our Team.

Kind regards,
Daniel

**Comment by Jenefer-Monroe** [comment]
The issue is not with the CoE package no. Sorry for any confusion, the comment from @shmcarth is about the product itself.

**Comment by zellos13** [comment]
> Collaborator

What Product exactly? Sorry if that is a stupid question - do they mean Power Platform or what?

And how do I resolve this issue myself? 

This is what I was able to find:

I tried with a classic designer to run a two-stage CoE Upgrade on both the May 2024 and November 2023 versions.

It didn't work on any of them, and the problem is the same. However, I was able to retrieve some information about the component from the classic designer. Maybe you will be able to provide me with the information.

After clicking Apply Upgrade error, it popped up.

![image](https://github.com/user-attachments/assets/850f88f1-b826-408f-96c2-1695a114309f)

But in the classic designer, I could click the "solution layers" phrase, and I saw this:

![image](https://github.com/user-attachments/assets/6a80fde7-d8e0-4ee7-86f4-a23cf790199d)

After clicking the GNCoECustomization solution layer, I saw:

![image](https://github.com/user-attachments/assets/9f339ff4-20dd-4f82-ba57-dd7a82cb3605)

And All Properties shows:

![image](https://github.com/user-attachments/assets/02ec6fb4-0ecd-4ea2-b853-df35bcf811a6)
![image](https://github.com/user-attachments/assets/54d842d0-7025-4325-96c2-a090bb803890)

There is something wrong with the environmental entity, as I understand. Do you have any idea what it is and how to resolve it?

**Comment by Jenefer-Monroe** [comment]
Yes the Power Platform itself is what has the issue and which is being addressed by the product team. 
Will have to defer to @shmcarth for the rest of your inquiry, this looks like great additional help for the team to see.

**Comment by zellos13** [comment]
Alright. I would appreciate a few words of comment; when can we expect that fix so we can test it?

**Comment by zellos13** [comment]
Hi @Jenefer-Monroe @shmcarth 

Do you happen to have any updates on this case, by chance? When can I upgrade the CoE after implementing the fix you are talking about?

We are stuck with the upgrade and I would appreciate any help.

**Comment by zellos13** [comment]
Hello,

To help everyone else that encounters this issue I am sharing the information on how I resolved it.

I removed the 'CustomControlDefaultConfig' record with id 'b3a6b6fb-09c9-4b89-a71f-b8a623be4188' from the table.

How:

I used XRM Toolbox and this query with a FetchXml Tester to find a record in my environment:

![image](https://github.com/user-attachments/assets/5ba1274d-b9c3-45ad-8cdc-9a516a2c838f)

`<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="false" >
    <entity name="customcontroldefaultconfig" >
        <filter type="and" >
            <condition attribute="customcontroldefaultconfigid" operator="eq" value="{b3a6b6fb-09c9-4b89-a71f-b8a623be4188}" />
        </filter>
    </entity>
</fetch>`

Then, I used the Bulk Delete Tool in XRM Toolbox to remove this record:

![image](https://github.com/user-attachments/assets/c7fa1ace-3e20-48ce-b015-39714ec95817)

It was removed, and the problem disappeared.

I hope you will find it useful.

**Comment by Jenefer-Monroe** [comment]
Amazing! Thank you for the assistance!

**Comment by harini-2-y** [comment]
There is no further investigation is required as an issue is resolved. Now closing this ticket.
